packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -73,44 +96,51 @@ const Tooltip: React.FC<TooltipOwnProps> = (props) => {
 
   // sync state attribute with using state machine subscription
   React.useEffect(() => {
-    const unsubscribe = stateMachine.subscribe(({ state, previousState }) => {
-      if (state === 'open') {
-        if (previousState === 'waitingForRest') {
-          setStateAttribute('delayed-open');
-        }
-        if (previousState === 'checkingIfShouldSkipRestThreshold' || previousState === 'closed') {
-          setStateAttribute('instant-open');
+    const unsubscribe = stateMachine.subscribe(({ state, context }) => {
+      if (context.id === contentId) {
+        if (state === 'open') {
+          setStateAttribute(context.delayed ? 'delayed-open' : 'instant-open');
+        } else if (state === 'closed') {
+          setStateAttribute('closed');
         }
       }
-      if (state === 'closed') {
-        setStateAttribute('closed');
-      }
     });
 
     return unsubscribe;
-  }, []);
+  }, [contentId]);
+
+  const handleFocus = React.useCallback(() => stateMachine.send({ type: 'FOCUS', id: contentId }), [
+    contentId,
+  ]);
+  const handleOpen = React.useCallback(
+    () => stateMachine.send({ type: 'OPEN', id: contentId, delayDuration }),
+    [contentId, delayDuration]
+  );
+  const handleClose = React.useCallback(
+    () => stateMachine.send({ type: 'CLOSE', id: contentId, skipDelayDuration }),
+    [skipDelayDuration, contentId]
+  );
 
   // send transition if the component unmounts
-  React.useEffect(() => {
-    return () => {
-      stateMachine.send({ type: 'UNMOUNT', id: contentId });
-    };
-  }, [contentId]);
+  React.useEffect(() => () => handleClose(), [handleClose]);
 
   // if we're controlling the component
   // put the state machine in the appropriate state
   useLayoutEffect(() => {
     if (openProp === true) {
-      stateMachine.send({ type: 'FOCUS', id: contentId });
+      handleFocus();
     }
-  }, [contentId, openProp]);
+  }, [handleFocus, openProp]);
```

Comment:
This part was weird for me. Feels like we should be calling `OPEN` here but obviously we aren't because we don't want the delay. That reasoning is super implicit atm though. I'm wondering if we should do something like this instead:

```jsx
useLayoutEffect(() => {
	if (openProp === true) {
		stateMachine.send({ type: 'OPEN', id: contentId, delayDuration: 0 });
	}
}, [openProp, contentId]);
```

🤔  ?

Reply:
Yeah I don't love this bit either.
I'm down with your suggestion, feels more explicit to what we want to do. 👍

I'll update!